apis/cpi/v1alpha1/vspherecpiconfig_types.go

```diff
@@ -13,25 +13,39 @@ type VSphereCPIConfigSpec struct {
 	VSphereCPI VSphereCPI `json:"vsphereCPI"`
 }
 
-type NSXTRoute struct {
+type NSXTRouteConfig struct {
 	// NSX-T T0/T1 logical router path
 	// +kubebuilder:validation:Optional
-	RouterPath string `json:"routerPath"`
+	RouterPath *string `json:"routerPath"`
+}
 
-	// Cluster CIDR
+type VMNetwork struct {
+	// Internal VM network name
+	// +kubebuilder:validation:Optional
+	Internal *string `json:"internal"`
+
+	// External VM network name
 	// +kubebuilder:validation:Optional
-	ClusterCidr string `json:"clusterCidr"`
+	External *string `json:"external"`
+
+	// Internal VM network CIDR to be excluded
+	// +kubebuilder:validation:Optional
+	ExcludeInternalSubnetCidr *string `json:"excludeInternalSubnetCidr"`
+
+	// External VM network CIDR to be excluded
+	// +kubebuilder:validation:Optional
+	ExcludeExternalSubnetCidr *string `json:"excludeExternalSubnetCidr"`
 }
 
 type NSXT struct {
 	// A flag that enables pod routing
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:default:=false
-	PodRoutingEnabled bool `json:"podRoutingEnabled"`
+	PodRoutingEnabled *bool `json:"podRoutingEnabled"`
```

Comment:
For optional one, you need omitempty in the json body as well.

Reply:
ok, let me add it